packages/rabbitmq/src/amqp/connection.ts

```diff
@@ -823,12 +1011,20 @@ export class AmqpConnection {
         consumer.rpcOptions,
         consumer.channel
       );
-    } else {
-      newConsumerTag = await this.setupSubscriberChannel<T>(
+    } else if (consumer.type === 'subscribe') {
+      newConsumerTag = await this.setupSingleSubscriberChannel<T>(
+        consumer.handler,
+        consumer.msgOptions,
+        consumer.channel
+      );
+    } else if (consumer.type === 'subscribe-batch') {
+      newConsumerTag = await this.setupBatchSubscriberChannel<T>(
         consumer.handler,
         consumer.msgOptions,
         consumer.channel
       );
+    } else {
+      throw new Error('Unable to resume consumer, unexpected consumer type.');
```

Comment:
I expect more contextual data as part of this error/exception otherwise the developer will not be able to quickly find the issue